com.unity.render-pipelines.core/Runtime/XR/XRPass.cs

```diff
@@ -0,0 +1,262 @@
+        /// Index of the pass inside the frame.
+        /// </summary>
+        public int multipassId { get; private set; }
+
+        /// <summary>
+        /// Index used for culling. It can be shared between multiple passes.
+        /// </summary>
+        public int cullingPassId { get; private set; }
+
+        /// <summary>
+        /// Destination render target.
+        /// </summary>
+        public RenderTargetIdentifier renderTarget { get; private set; }
+
+        /// <summary>
+        /// Destination render target descriptor.
+        /// </summary>
+        public RenderTextureDescriptor renderTargetDesc { get; private set; }
+
+        /// <summary>
+        /// Parameters used for culling.
+        /// </summary>
+        public ScriptableCullingParameters cullingParams { get; private set; }
+
+        /// <summary>
+        /// Returns the number of views inside this pass.
+        /// </summary>
+        public int viewCount { get => m_Views.Count; }
+
+        /// <summary>
+        /// If true, the render pipeline is expected to use single-pass techniques to save CPU time.
+        /// </summary>
+        public bool singlePassEnabled { get => viewCount > 1; }
+
+        /// <summary>
+        /// Returns the projection matrix for a given view.
+        /// </summary>
+        /// <param name="viewIndex"></param>
+        public Matrix4x4 GetProjMatrix(int viewIndex = 0)
+        {
+            return m_Views[viewIndex].projMatrix;
+        }
+
+        /// <summary>
+        /// Returns the view matrix for a given view.
+        /// </summary>
+        /// <param name="viewIndex"></param>
+        public Matrix4x4 GetViewMatrix(int viewIndex = 0)
+        {
+            return m_Views[viewIndex].viewMatrix;
+        }
+
+        /// <summary>
+        /// Returns the viewport for a given view.
+        /// </summary>
+        /// <param name="viewIndex"></param>
+        public Rect GetViewport(int viewIndex = 0)
+        {
+            return m_Views[viewIndex].viewport;
+        }
+
+        /// <summary>
+        /// Returns the occlusion mesh for a given view.
+        /// </summary>
+        /// <param name="viewIndex"></param>
+        public Mesh GetOcclusionMesh(int viewIndex = 0)
+        {
+            return m_Views[viewIndex].occlusionMesh;
+        }
+
+        /// <summary>
+        /// Returns the destination slice index (for texture array) for a given view.
+        /// </summary>
+        /// <param name="viewIndex"></param>
+        public int GetTextureArraySlice(int viewIndex = 0)
+        {
+            return m_Views[viewIndex].textureArraySlice;
+        }
+
+        /// <summary>
+        /// Queue up render commands to enable single-pass techniques.
+        /// Note: depending on the platform and settings, either single-pass instancing or the multiview extension will be used.
+        /// </summary>
+        /// <param name="cmd"></param>
+        public void StartSinglePass(CommandBuffer cmd)
```

Comment:
Any reason why we prefer to check for enabled keyword instead of #if ENABLE_VR && ENABLE_XR_MODULE?

Reply:
It's 2 different mechanisms:
- the runtime check + settings is done with xr.enabled
- the conditional compilation is just a way to strip out the XR code on platform who do not support XR, but even if the platform supports XR, it does not imply XR is used or active for this particular frame